presto-spi/src/main/java/com/facebook/presto/spi/eventlistener/PlanOptimizerInformation.java

```diff
@@ -29,16 +29,20 @@ public class PlanOptimizerInformation
     // For optimizers which are not enabled. True if the query matches the pattern of the optimizer and could be applied.
     // False if cannot be applied. Empty if information not available.
     private final Optional<Boolean> optimizerApplicable;
+    // True if optimizer encounter failures (for example timeout etc.), false if no failures, empty if information not available.
+    private final Optional<Boolean> optimizerFailure;
 
     @JsonCreator
     public PlanOptimizerInformation(
             @JsonProperty("optimizerName") String optimizerName,
             @JsonProperty("optimizerTriggered") boolean optimizerTriggered,
-            @JsonProperty("optimizerApplicable") Optional<Boolean> optimizerApplicable)
+            @JsonProperty("optimizerApplicable") Optional<Boolean> optimizerApplicable,
+            @JsonProperty("optimizerFailure") Optional<Boolean> optimizerFailure)
     {
         this.optimizerName = requireNonNull(optimizerName, "optimizerName is null");
         this.optimizerTriggered = requireNonNull(optimizerTriggered, "optimizerTriggered is null");
         this.optimizerApplicable = requireNonNull(optimizerApplicable, "optimizerApplicable is null");
+        this.optimizerFailure = requireNonNull(optimizerFailure, "optimizerApplicable is null");
```

Comment:
that's cool!
I think another use case for this field could be tracking bugs in disabled optimizers: when checking if an optimizer is applicable (when verbose_optimizer_info_enabled=true) I added a try/block to prevent the main loop crashing with a buggy (but disabled) optimizer: instead of silently ignoring this, we can record that we found a bug for an applicable optimizer (see PlanOptimizer::isApplicable)